pybamm/simulation.py

```diff
@@ -170,44 +170,79 @@ def set_up_experiment(self, model, experiment):
         self._experiment_inputs = []
         self._experiment_times = []
         for op, events in zip(experiment.operating_conditions, experiment.events):
-            if op[1] in ["A", "C"]:
-                # Update inputs for constant current
+            if isinstance(op[0], np.ndarray):
+                # If ndarray is recived from, create interpolant
+                # create interpolant
+                timescale = self._parameter_values.evaluate(model.timescale)
+                drive_cycle_interpolant = pybamm.Interpolant(
+                    op[0][:, 0], op[0][:, 1], timescale * pybamm.t
```

Comment:
The last input here is the time that is evaluated by the interpolant. You could do the following:
```python3
drive_cycle_interpolant = pybamm.Interpolant(
                    op[0][:, 0], op[0][:, 1], timescale * pybamm.t - pybamm.InputParameter("start time")
)
```
and pass the appropriate "start time" to the model inputs when solving the model at each step